frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import Button from 'patternfly-react/dist/esm/components/Button/Button';
+import CatalogItemHeader from 'patternfly-react-extensions/dist/esm/components/CatalogItemHeader/CatalogItemHeader';
+import { FieldLevelHelp } from 'patternfly-react';
+import { Dropdown, Firehose } from '../utils';
+import { ProjectModel } from '../../models';
+
+const getItems = (props) => {
+  const {namespace, loaded} = props;
+  if (!loaded || !namespace) {
+    return {};
+  }
+  let items = [];
+  _.forEach(namespace.data, ns => {
+    const name = _.get(ns, 'metadata.name', false);
+    if (name) {
+      items.push(name);
+    }
+  });
+  items = _.sortBy(items);
+  return _.zipObject(items, items);
+};
+
+const NamespaceDropdown = (props) => {
+  // Some namespaces are too long to fit in the dropdown
+  // when the screen is small
+  const { title, onChange, id } = props;
+  const items = getItems(props);
+  return <Dropdown
+    title={title}
+    id={id}
+    items={items}
+    dropDownClassName="dropdown--full-width"
+    onChange={onChange}
+  />;
+}
+
+class DropdownElement extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    const { title, id, items, onChange, help } = this.props;
+    return (
+      <div className="co-marketplace-subscribe--dropdownelement">
+        {title}
+        {help && <FieldLevelHelp content={help} />}
+        <br />
+        {id === 'namespace'
+        ?<Firehose resources={[{ kind: ProjectModel.kind, prop: 'namespace', isList: true }]}>
+          <NamespaceDropdown
+            title={title}
+            id={id}
+            onChange={onChange}
+          />
+        </Firehose>
+        :<Dropdown
+          title={title}
+          id={id}
+          items={items}
+          dropDownClassName="dropdown--full-width"
+          onChange={onChange}
+        />}
+      </div>
+    );
+  }
+}
+class AdminSubscribe extends React.Component {
```

Comment:
Can we name the file after the class (admin-subscribe.jsx)? It gets confusing when they differ.